### PR TITLE
Added syntax highlighting for .odin files

### PR DESCRIPTION
**Pull request by leidegre**
This PR adds syntax highlighting for .odin files ([Odin programming language](https://odin-lang.org/)) and is based on the `txti_token_array_from_string__cpp` function.

Key differences are:

- Set of keywords
- `#meta` is treated like an identifier
- Identifier with leading Upper_Case is treated as a type name

I added a `TXTI_TokenKind_Type` and mapped that to `DF_ThemeColor_CodeType`.

Example

image

Let me know if you want me to adjust things. 

I tried out the 4coder editor and it was a joyful experience, I think I will try to use this more.

### CODE REVIEW

File: src/txti/txti.c [hunk 1]
```diff
@@ -421,6 +425,248 @@ txti_token_array_from_string__cpp(Arena *arena, U64 *bytes_processed_counter, St
...
+            str8_lit_comp("else"),
+            str8_lit_comp("enum"),
+            str8_lit_comp("fallthrough"),
+            str8_lit_comp("for"),
+            str8_lit_comp("foreign"),
+            str8_lit_comp("if"),
+            str8_lit_comp("if"),
+            str8_lit_comp("in"),
+            str8_lit_comp("inline"),
+            str8_lit_comp("inline"),
```
Comment (gingerBill) [inline src/txti/txti.c:615]: `inline` is not a keyword

File: src/txti/txti.c [hunk 2]
```diff
@@ -421,6 +425,248 @@ txti_token_array_from_string__cpp(Arena *arena, U64 *bytes_processed_counter, St
...
+            str8_lit_comp("distinct"),
+            str8_lit_comp("do"),
+            str8_lit_comp("dynamic"),
+            str8_lit_comp("else"),
+            str8_lit_comp("enum"),
+            str8_lit_comp("fallthrough"),
+            str8_lit_comp("for"),
+            str8_lit_comp("foreign"),
+            str8_lit_comp("if"),
+            str8_lit_comp("if"),
```
Comment (gingerBill) [inline src/txti/txti.c:612]: Repetition of keyword

File: src/txti/txti.c [hunk 3]
```diff
@@ -421,6 +425,248 @@ txti_token_array_from_string__cpp(Arena *arena, U64 *bytes_processed_counter, St
...
+            str8_lit_comp("in"),
+            str8_lit_comp("inline"),
+            str8_lit_comp("inline"),
+            str8_lit_comp("map"),
+            str8_lit_comp("matrix"),
+            str8_lit_comp("no_inline"),
```
Comment (gingerBill) [inline src/txti/txti.c:618]: `no_inline` is not a keyword

File: src/txti/txti.c [hunk 4]
```diff
@@ -69,6 +69,10 @@ txti_lang_kind_from_extension(String8 extension)
   {
     kind = TXTI_LangKind_CPlusPlus;
   }
+  else if(str8_match(extension, str8_lit("odin"), StringMatchFlag_CaseInsensitive))
```
Comment (gingerBill) [inline src/txti/txti.c:72]: Odin only supports `.odin` in lowercase, not any case.

File: src/txti/txti.c (hunk 3 above)
Comment (leidegre) [reply src/txti/txti.c]: Fixed

File: src/txti/txti.c (hunk 2 above)
Comment (leidegre) [reply src/txti/txti.c]: Fixed

File: src/txti/txti.c (hunk 1 above)
Comment (leidegre) [reply src/txti/txti.c]: Fixed

**Comment by leidegre** [comment]
@gingerBill thanks for taking a look. I only started my Odin journey last week and really appreciate the help.

File: src/txti/txti.c [hunk 5]
```diff
@@ -421,6 +425,244 @@ txti_token_array_from_string__cpp(Arena *arena, U64 *bytes_processed_counter, St
...
+        TXTI_Token token = {active_token_kind, r1u64(active_token_start_idx, idx+ender_pad)};
+        active_token_kind = TXTI_TokenKind_Null;
+        
+        // rjf: identifier -> keyword in special cases
+        if(token.kind == TXTI_TokenKind_Identifier)
+        {
+          read_only local_persist String8 cpp_keywords[] =
+          {
+            str8_lit_comp("asm"),
+            str8_lit_comp("auto_case"),
```
[the remark this answers is not in the document]
Comment (leidegre) [reply src/txti/txti.c]: Thanks!

**Comment by ryanfleury** [comment]
Held off on this because I was stalled on some other stuff, and I needed to deduplicate stuff in the `TXTI` and `TXT` layers. Thanks for looking into this. I've merged this by hand in 4bc82db69eed8b563ab4ea5f4a86e56bb783d128, currently just on the `dev` branch. This isn't full Odin support but will enable the bare minimum stuff - syntax highlighting, go-to-definition, hover-eval, and so on (there may still be holes with such features as they depart from being C-style, e.g. `.` operator for symbol names, and so on). This will be in the next release.

**Pull request closed without merging**